cli/cli_app.go

```diff
@@ -878,81 +884,92 @@ func remoteStateNeedsInit(remoteState *remote.RemoteState, terragruntOptions *op
 	return false, nil
 }
 
-// planAll prints the plans from all configuration in a stack, in the order
-// specified in the terraform_remote_state dependencies
-func planAll(terragruntOptions *options.TerragruntOptions) error {
+// runAll runs the provided terraform command against all the modules that are found in the directory tree.
+func runAll(terragruntOptions *options.TerragruntOptions) error {
 	stack, err := configstack.FindStackInSubfolders(terragruntOptions)
 	if err != nil {
 		return err
 	}
 
 	terragruntOptions.Logger.Printf("%s", stack.String())
-	return stack.Plan(terragruntOptions)
-}
 
-// Spin up an entire "stack" by running 'terragrunt apply' in each subfolder, processing them in the right order based
-// on terraform_remote_state dependencies.
-func applyAll(terragruntOptions *options.TerragruntOptions) error {
-	stack, err := configstack.FindStackInSubfolders(terragruntOptions)
-	if err != nil {
-		return err
+	var prompt string
+	switch terragruntOptions.TerraformCommand {
+	case "apply":
+		prompt = "Are you sure you want to run 'terragrunt apply' in each folder of the stack described above?"
+	case "destroy":
+		prompt = "WARNING: Are you sure you want to run `terragrunt destroy` in each folder of the stack described above? There is no undo!"
```

Comment:
Any reason we only prompt for `apply` and `destroy`? Is it because they might result in changes? If so, what about `import`, `taint`, `untaint`, and `state`? Also, should some command explicitly be blocked from `run-all`? E.g., `console`, `force-unlock`, `login`, `logout`, and a bunch of others don't make sense to combine with `run-all`. Or, alternatively, we could have a set of allowed commands, and block all others.

Reply:
Good point. I opted to specify the list of commands to explicitly disallow, since I think it is easier to maintain a disallow list than it is to maintain an allow list.

4ba280428beb919a44b8f83607ae1c14496c6c6c

Reply:
Nice 👍 

One more sanity check: we now disallow a few commands... Are there any others beyond `apply` and `destroy` that merit a prompt from Terragrunt?

Reply:
Added `state` to prompt: 7f4d53a . I think that's the only command left?